module/batch-action/src/Infrastructure/Persistence/Query/DbalBatchActionQuery.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Infrastructure\Persistence\Query;
+
+use Ergonode\BatchAction\Domain\Entity\BatchActionId;
+use Ergonode\BatchAction\Domain\Model\BatchActionInformationModel;
+use Ergonode\BatchAction\Domain\Query\BatchActionQueryInterface;
+use Doctrine\DBAL\Connection;
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionType;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\SharedKernel\Domain\AggregateId;
+use Ergonode\BatchAction\Domain\Model\BatchActionEntryModel;
+
+class DbalBatchActionQuery implements BatchActionQueryInterface
+{
+    private Connection $connection;
+
+    private TranslatorInterface $translator;
+
+    public function __construct(Connection $connection, TranslatorInterface $translator)
+    {
+        $this->connection = $connection;
+        $this->translator = $translator;
+    }
+
+    public function getInformation(BatchActionId $id, Language $language): BatchActionInformationModel
+    {
+        $qb = $this->connection->createQueryBuilder();
+
+        $record = $qb
+            ->select('id, created_at, type')
+            ->addSelect('(SELECT count(*) FROM batch_action_entry WHERE batch_action_id = id) AS all')
+            ->addSelect('(SELECT count(*) FROM batch_action_entry WHERE batch_action_id = id 
+            AND success IS NOT NULL) AS processed')
+            ->addSelect('(SELECT max(processed_at) FROM batch_action_entry WHERE batch_action_id = id) 
```

Comment:
This will give you the "current" last processed date, not necessarily the end date of the entire process if I'm reading correctly

Reply:
i use/return this information only if count of proceed entities is equal  count of all entries, in that case this date is date of last processed element.

Reply:
ahhh now I see it 👍🏻